### PR TITLE
feat(snackbar): add feature targeting for styles

### PR DESCRIPTION
**Pull request by crisbeto**
Adds support for feature targeting to the `mdc-snackbar` styles.

Relates to #4227.

### CODE REVIEW

**Comment by crisbeto** [comment]
Here's a comparison between the before and after styles. A couple of lines had to be moved around so that they'd fit into the mixins.

[before.css.txt](https://github.com/material-components/material-components-web/files/3360464/before.css.txt)
[after.css.txt](https://github.com/material-components/material-components-web/files/3360465/after.css.txt)

File: packages/mdc-snackbar/_mixins.scss [hunk 1]
```diff
@@ -20,73 +20,269 @@
...
 @import "./variables";
 
-@mixin mdc-snackbar-fill-color($color) {
+@mixin mdc-snackbar-core-styles($query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+  $feat-animation: mdc-feature-create-target($query, animation);
+
+  // postcss-bem-linter: define snackbar
+  .mdc-snackbar {
+    @include mdc-snackbar-z-index($mdc-snackbar-z-index, $query);
```
Comment (abhiomkar) [inline packages/mdc-snackbar/_mixins.scss]: Let's name all `$query` params in this file when including mixins. 

```scss
@include mdc-snackbar-z-index($mdc-snackbar-z-index, $query: $query);
```

File: packages/mdc-snackbar/_mixins.scss [hunk 2]
```diff
@@ -20,73 +20,269 @@
 // THE SOFTWARE.
 //
 
+@import "@material/feature-targeting/functions";
```
Comment (abhiomkar) [inline packages/mdc-snackbar/_mixins.scss]: Update mdc-snackbar/package.json with @material/feature-targeting dependency.

**Comment by crisbeto** [comment]
Thank you for the feedback @abhiomkar, it should all be addressed now. Can you take another look?

**Pull request merged**